tests/collections/asr/test_asr_multitask_model_bpe.py

```diff
@@ -275,6 +276,51 @@ def test_decoding_change(self, asr_model):
         assert isinstance(asr_model.decoding.decoding, beam_decode.TransformerAEDBeamInfer)
         assert asr_model.decoding.decoding.search_type == "default"
 
+    @pytest.mark.unit
+    def test_prompt_change(self, asr_model):
+        assert asr_model.prompt_format == 'canary'
+        assert isinstance(asr_model.prompt, CanaryPromptFormatter)
+
+        # Default change prompt
+        asr_model.change_prompt()
+        assert asr_model.cfg.prompt_defaults is None
+
+        prompt_defaults = asr_model.prompt.get_default_dialog_slots()
+        prompt_defaults[0]['slots']['pnc'] = 'no'
+        asr_model.change_prompt(prompt_defaults=prompt_defaults)
+
+        assert asr_model.cfg.prompt_defaults[0]['slots']['pnc'] == 'no'
+
+    @pytest.mark.unit
+    def test_prompt_change_subclass(self, asr_model):
+        assert asr_model.prompt_format == 'canary'
+        assert isinstance(asr_model.prompt, CanaryPromptFormatter)
+
+        class CanaryPromptFormatterSubclass(CanaryPromptFormatter):
+            NAME = "canary2"
+
+        # Default change prompt
+        asr_model.change_prompt()
```

Comment:
If we're allowing no arguments, I wonder whether `reset_prompt` is not a more appropriate name. As is this is slightly confusing.

Reply:
We have similar semantics for change decoding strategy. Calling it with none sets up the original config with all defaults

Reply:
Turns out i cant do that cause all the default values are None and then it messes up if user doesn't explicitly override. I really wish we had default values (from config or somewhere). For now ill just leave None prompt_defaults as a No-op

Reply:
I have an idea. How about adding this to `CanaryPromptFormatter`:

```
def __init__(self, tokenizer, defaults) -> None:
    if defaults is None:
        defaults = ...  # appropriate values for the released canary-1b checkpoint
    super().__init__(tokenizer, defaults)
```

Reply:
Yes might be a good idea but lets do it in another PR cause i don't actually know all the defaults accepted by canary